daily_post.py

```diff
@@ -10,29 +12,74 @@
 import daily_markov
 
 
-def get_weather() -> str:
-    resp = requests.get(
-        'http://forecast.weather.gov/MapClick.php?lat=47.62&lon=-122.36&unit=0&lg=english&FcstType=text&TextType=1')
-    doc = resp.content
-    soup_doc = BeautifulSoup(doc, 'html.parser')
-    weather_info = soup_doc.find_all('table')[1]
-    reddit_comment = ''
-    if weather_info.find_all('div'):
-        advisories = weather_info.div.extract().find_all('a')
-        reddit_comment += '* Advisories:\n'
-        for advis in advisories:
-            reddit_comment += ' * [{}](http://forecast.weather.gov/{})\n'.format(
-                advis.span.string, advis.get('href'))
-    forecast = str(weather_info).split("<br/>\n<br/>")
-    top_three = '<br/>\n<br/>'.join(str(x) for x in forecast[1:5])
-    top_three_doc = BeautifulSoup(top_three, 'html.parser')
-    reddit_comment += '* {}'.format(str.replace(
-        top_three_doc.text, '\n\n', "\n* ")[1:-1])
-
-    return reddit_comment
-
-
-def gen_post(weekday: int) -> str:
+DISCOVERY_URL = 'https://api.weather.gov/points/47.62,-122.36'
+FORECAST_GRID_DATA_URL = 'https://api.weather.gov/gridpoints/SEW/123,68'
+FORECAST_URL = 'https://api.weather.gov/gridpoints/SEW/123,68/forecast'
+FORECAST_HOURLY_URL = 'https://api.weather.gov/gridpoints/SEW/123,68/forecast/hourly'
+
+
+class Emojis:
+    NewMoon = '\U0001F311'
+    Sunny = '\u2600'
+    Cloudy = '\u2601'
+    PartlyCloudy = '\u26C5'
+    RainAndLightning = '\u26C8'
+    Rain = '\U0001F327'
+    Snow = '\U0001F328'
+    Lightning = '\U0001F329'
+    Fog = '\U0001F32B'
+
+
+def _convert_time_str_to_datetime(time: str) -> datetime.datetime:
+    time = re.sub(r'([-+]\d\d):(\d\d)', r'\1\2', time)
+    return datetime.datetime.strptime(time, '%Y-%m-%dT%H:%M:%S%z')
+
+
+def get_weather_emoji(period: Dict[str, Any]) -> Optional[str]:
+    # only get the first phrase in the short forecast
+    match = re.fullmatch(r'((?:[A-Z]\w+ )+[A-Z]\w+)', period['shortForecast'])
```

Comment:
Strings are indexed, so we can just get `period['shortForecast'][0]` -- unless there's something else you're wishing to accomplish with this?

Reply:
The regex captures only the first part of short forecasts with two parts, like `Mostly Cloudy then Chance Rain Showers`

Reply:
basically grab words up to the lowercased word

Reply:
Hrmmm ok, seems fine for now. Seems chancey the API could change and break royally, here, but simple enough to find out when it does.

Reply:
such is the curse of parsing text in undocumented formats